packages/shared/src/components/profile/ProfileForm.tsx

```diff
@@ -97,37 +87,38 @@ export default function ProfileForm({
 
   const onSubmit = async (event: FormEvent): Promise<void> => {
     event.preventDefault();
-    setDisableSubmit?.(true);
-    const data = formToJson<UserProfile>(formRef.current, {
-      name: '',
-      email: '',
-      username: '',
-      timezone: userTimeZone,
-    });
 
-    const res = await updateProfile(data);
-    if ('error' in res) {
-      if ('code' in res && res.code === 1) {
-        if (res.field === 'email') {
-          setEmailHint('This email is already used');
-        } else if (res.field === 'username') {
-          setUsernameHint('This username is already taken');
-        } else if (res.field === 'twitter') {
-          setTwitterHint('This Twitter handle is already used');
-        } else if (res.field === 'github') {
-          setGithubHint('This GitHub handle is already used');
-        } else if (res.field === 'hashnode') {
-          setGithubHint('This Hashnode handle is already used');
+    if (formRef.current.checkValidity()) {
```

Comment:
What happens in case the form is not valid? Can you attach a print screen with an example please?

Reply:
You basically get these native HTML errors besides the ones we have set.

<img width="625" alt="Screenshot 2022-06-24 at 17 47 14" src="https://user-images.githubusercontent.com/554874/175571066-55d87320-8671-4771-8f61-8ada09d63e93.png">

<img width="611" alt="Screenshot 2022-06-24 at 17 46 58" src="https://user-images.githubusercontent.com/554874/175571062-07e76706-1f75-4dcb-a860-68a06a0f2e20.png">

Reply:
Not the best UI for this but given it's about to be changed soon that's ok